### PR TITLE
nlopt-ocaml: there is no valid reason for the ocaml-version upper bound

### PR DESCRIPTION
**Pull request by UnixJunkie**
according to my compilation tests

### CODE REVIEW

**Comment by UnixJunkie** [comment]
related to https://github.com/mkur/nlopt-ocaml/issues/2

**Comment by UnixJunkie** [comment]
CI tests look OK to me.
However, some brew build seems to have stalled on the mac.
```
==> Patching
patching file gcc/graphite.h
==> ../configure --build=x86_64-apple-darwin15.6.0 --prefix=/usr/local/Cellar/gc
==> make
Wed Jan 23 04:13:11 UTC 2019 - building ...
Wed Jan 23 04:19:11 UTC 2019 - building ...
Wed Jan 23 04:25:11 UTC 2019 - building ...
Wed Jan 23 04:31:11 UTC 2019 - building ...
The job exceeded the maximum time limit for jobs, and has been terminated.
```

**Comment by kit-ty-kate** [comment]
mmh there is a problem here, the version is 0.5.2 but the archive pulled is 0.5.1. Is that just a change in the opam file and not in the source code? In that case, changing the opam file of the version itself is a much better idea

File: packages/nlopt-ocaml/nlopt-ocaml.0.5.2/opam [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+license: "LGPL-2.1+ with OCaml linking exception"
+build: [
+  ["oasis" "setup"]
+  ["ocaml" "setup.ml" "-configure" "--prefix" prefix]
+  ["ocaml" "setup.ml" "-build"]
+]
+remove: [["ocamlfind" "remove" "nlopt"]]
+depends: [
+  "ocaml" {>= "3.12"}
+  "ocamlfind"
```
Comment (kit-ty-kate) [inline packages/nlopt-ocaml/nlopt-ocaml.0.5.2/opam]:
```suggestion
  "ocamlfind" {build}
```
also the source code doesn't seem to make any use of `ocamlfind` outside the build phase. Could you add the the `{build}` attribute if appropriate?

**Comment by UnixJunkie** [comment]
ok, superseded by
https://github.com/ocaml/opam-repository/pull/13315

**Pull request closed without merging**